File: tobas_setup_assistant/src/tobas_setup_assistant/parameter_getters/check_box.py
from overrides import override
from typing import Optional
from PyQt5.QtCore import pyqtSignal, pyqtSlot
from PyQt5.QtWidgets import QCheckBox

from .base import ParamGetterWidget


class ParamGetterWidget_CheckBox(ParamGetterWidget):
    toggled = pyqtSignal(bool)

    def __init__(
        self,
        param_name: str,
        description_text: Optional[str] = None,
        check_box_text: Optional[str] = None,
        default: bool = False,
    ) -> None:
        super().__init__(param_name, description_text)

        self._box = QCheckBox(check_box_text)
        self._box.setChecked(default)
        self._box.toggled.connect(self._on_toggled)
        self._rows.addWidget(self._box)

    @override
    def get(self) -> bool:
        return self._box.isChecked()

    @override
    def set(self, src: bool) -> None:
        self._box.setChecked(src)

    @pyqtSlot()
    def _on_toggled(self) -> None:
        self.toggled.emit(self.get())

File: tobas_gui/tobas_setup_assistant/src/param_getters/check_box.cpp
#include "tobas_setup_assistant/param_getters/check_box.hpp"

namespace gui
{
namespace setup_assistant
{
ParamGetterWidget_CheckBox::ParamGetterWidget_CheckBox(const QString& param_name, const QString& description_text)
  : super(param_name, description_text)
{
  box_ = new QCheckBox();
  connect(box_, &QCheckBox::toggled, this, &self::onToggled);
  rows_->addWidget(box_);
}

bool ParamGetterWidget_CheckBox::getValue() const
{
  return box_->isChecked();
}

bool ParamGetterWidget_CheckBox::setValue(const bool& src)
{
  box_->setChecked(src);
  return true;
}

void ParamGetterWidget_CheckBox::setText(const QString& text)
{
  box_->setText(text);
}

void ParamGetterWidget_CheckBox::onToggled(bool checked)
{
  Q_EMIT toggled(checked);
}
}  // namespace setup_assistant
}  // namespace gui
